ledger/src/shred/wire.rs

```diff
@@ -152,10 +151,8 @@ fn get_data_size(shred: &[u8]) -> Result<u16, Error> {
 #[inline]
 pub(crate) fn get_data(shred: &[u8]) -> Result<&[u8], Error> {
     match get_shred_variant(shred)? {
-        ShredVariant::LegacyCode => Err(Error::InvalidShredType),
-        ShredVariant::MerkleCode { .. } => Err(Error::InvalidShredType),
-        ShredVariant::LegacyData => {
-            shred::legacy::ShredData::get_data(shred, get_data_size(shred)?)
+        ShredVariant::LegacyCode | ShredVariant::LegacyData | ShredVariant::MerkleCode { .. } => {
+            Err(Error::InvalidShredType)
```

Comment:
Got here while reviewing https://github.com/anza-xyz/agave/pull/7570 - this PR effectively removed the ability to read data shreds right ? If so, I think we probably should have accompanied the change with a CHANGELOG entry. 

There may not be legacy shreds floating around on the public clusters, but people may still be doing stuff like loading old ledgers, and the CHANGELOG entry is a reminder that newer software cannot read older Blockstores that contain legacy shreds

Reply:
Yeah good point will make PR